internal/mode/static/telemetry/collector.go

```diff
@@ -0,0 +1,78 @@
+package telemetry
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/state"
+)
+
+type Collector interface {
+	CollectData() Data
```

Comment:
What if we name it DataCollector (so we now from the type what it deals with) and rename CollectData to Collect (we can infer that we collecting Data from the type name)?

Note: we can change DataCollector type name to DataCollectorImpl - it is not a big deal to have impl